test/blackbox-tests/test-cases/pkg/lock-directory-selection.t

```diff
@@ -0,0 +1,115 @@
+Test that dune can dynamically select a lockdir with a cond statement 
+
+  $ . ./helpers.sh
+
+  $ mkrepo
+
+  $ mkpkg arm64-only <<EOF
+  > install: [ "echo" "arm64-only" ]
+  > EOF
+
+  $ mkpkg linux-only <<EOF
+  > install: [ "echo" "linux-only" ]
+  > EOF
+
+  $ mkpkg macos-only <<EOF
+  > install: [ "echo" "macos-only" ]
+  > EOF
+
+  $ cat >dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path dune.macos.arm64.lock)
+  >  (repositories mock)
+  >  (solver_env
+  >   (arch arm64)
+  >   (os macos)))
+  > (lock_dir
+  >  (path dune.macos.lock)
+  >  (repositories mock)
+  >  (solver_env
+  >   (arch amd64)
+  >   (os macos)))
+  > (lock_dir
+  >  (path dune.linux.lock)
+  >  (repositories mock)
+  >  (solver_env
+  >   (arch amd64)
+  >   (os linux)))
+  > (lock_dir
+  >  (repositories mock))
+  > (repository
+  >  (name mock)
+  >  (source "file://$(pwd)/mock-opam-repository"))
+  > (context
+  >  (default
+  >   (lock_dir (cond
+  >    ((and (= %{architecture} arm64) (= %{system} macos)) dune.macos.arm64.lock)
```

Comment:
Do you think the extra pairs of paren is are necessary? It could be:

```
(cond
 (and (= %{architecture} arm64) (= %{system} macos)) dune.macos.arm64.lock
 ..)
```

So we make `cond` accept a list of an even number of elements. I know this isn't a how lisp does it, but clojure does it and it seems a little neater.

Reply:
I also like how clojure uses `:else` to mark the catch-all clause.

Reply:
I prefer to keep the extra parens as the conditions and lockfiles are more explicitly tied together by syntax. If we alternated interpreting terms as conditions and lockdirs then I'd worry about the types of syntax error that would become possible if dune accidentally interpreted a lockdir as a condition or vice versa.

I will add a catch-all clause.